R/revision_analysis.R

```diff
@@ -179,85 +176,84 @@ revision_analysis <- function(vintages,
 
     # Parametric analysis
 
-    quality_thresholds <- set_thresholds()
-    quality_thresholds_res <- list(jb = quality_thresholds$jb_res,
-                                   bp = quality_thresholds$bp_res,
-                                   white = quality_thresholds$white_res,
-                                   arch = quality_thresholds$arch_res)
+    res_thresholds <- list(jb = getOption("jb_res_threshold"),
+                           bp = getOption("bp_res_threshold"),
+                           white = getOption("white_res_threshold"),
+                           arch = getOption("arch_res_threshold"))
 
     ## I. Relevancy
 
     ### Theil tests (U1 and U2)
     U1 <- theil(vt, gap, na.zero)
     U2 <- theil2(vt, gap, na.zero)
     theil_infos <- theil_test_evaluator(U1, U2, N = ds["N", ], n_test = ncol(rv),
-                                        thr = list(u1 = quality_thresholds$theil_u1,
-                                                   u2 = quality_thresholds$theil_u2))
+                                        thr = list(u1 = getOption("theil_u1_threshold"),
+                                                   u2 = getOption("theil_u2_threshold")))
 
     ## II. Bias (mean and regression bias)
 
     ### T-test and Augmented T-test
     tat_test <- bias(rv, na.zero)
     tat_infos <- tat_test_evaluator(tat_test, is_log, n_test = ncol(rv),
-                                    thr = list(t = quality_thresholds$t,
-                                               at = quality_thresholds$augmented_t))
+                                    thr = list(t = getOption("t_threshold"),
+                                               at = getOption("augmented_t_threshold")))
 
     ### Slope and drift
     sd_test <- slope_and_drift(vtc, gap, na.zero)
     sd_infos <- sd_test_evaluator(sd_test, is_log, is_cointegrated, delta_diff,
                                   n_test = ncol(vtc[, -c(1:gap), drop = FALSE]),
-                                  thr = quality_thresholds$slope_and_drift,
-                                  thr_res = quality_thresholds_res)
+                                  thr = getOption("slope_and_drift_threshold"),
+                                  thr_res = res_thresholds)
```

Comment:
You should detail each threshold individually. This will make it easier for the user to set thresholds individually.

For example:

instead of:

``` r
res_thresholds <- list(
    jb = getOption("jb_res_threshold"),
    bp = getOption("bp_res_threshold"),
    white = getOption("white_res_threshold"),
    arch = getOption("arch_res_threshold")
)
 sd_test_evaluator(..., thr_res = res_thresholds)
```

You can write:

``` r
res_thresholds <- list(
    jb = getOption("jb_res_threshold"),
    bp = getOption("bp_res_threshold"),
    white = getOption("white_res_threshold"),
    arch = getOption("arch_res_threshold")
)
 sd_test_evaluator(..., 
    jb = getOption("jb_res_threshold"),
    bp = getOption("bp_res_threshold"),
    white = getOption("white_res_threshold"),
    arch = getOption("arch_res_threshold")
)
```

Maybe the best is to add an argument `...` to `eval_test()` instead of `threshold`.